web_src/js/markup/math.js

```diff
@@ -23,12 +23,14 @@ export async function renderMath() {
 
   for (const el of els) {
     const source = el.textContent;
-    const options = {display: el.classList.contains('display')};
+    const nodeName = el.classList.contains('display') ? 'p' : 'span';
 
     try {
-      const markup = katex.renderToString(source, options);
-      const tempEl = document.createElement(options.display ? 'p' : 'span');
-      tempEl.innerHTML = markup;
+      const tempEl = document.createElement(nodeName);
+      katex.render(source, tempEl, {
+        maxSize: 25,
+        maxExpand: 50,
```

Comment:
Why the maxSize?
Won't that break nested fractions?

Reply:
I took inspiration from the [GitLab config](https://gitlab.com/gitlab-org/gitlab/-/blob/master/app/assets/javascripts/behaviors/markdown/render_math.js) where they use 20 for both values. maxSize should as far as I understand limit the size of rendered elements only.

Reply:
Reference: https://github.com/go-gitea/gitea/issues/27812
Default value is 1000.

Reply:
And GitLab increased the value from 20 to 1000.
https://gitlab.com/gitlab-org/gitlab/-/commit/4d490b4bcfb7081c92550add80c6687b1870a176

Reply:
So, I guess then we can remove this parameter either completely, or we set it to its default value.